lib/federation.js

```diff
@@ -129,6 +129,7 @@ function getStubTypes (schemaDefinitions, isGateway) {
 
 function gatherDirectives (type) {
   let directives = []
+  /* istanbul ignore else seems that extensionASTNodes is always provided in graphql16+, we are keeping this check for safety */
```

Comment:
what is this comment doing?

Reply:
In graphql 15 `extensionASTNodes` could be either `null` or `Array`

Now it is only `Array` but I thought that was safer to keep the check.

I've added this comment to get the 100% coverage and explain why I didn't get rid of it.

Reply:
ok so this comment mixes an istanbul command with human-readable information.
I would recommend we keep the two separated, meaning, 2 comments